src/nbl/video/COpenGLCommandPool.h

```diff
@@ -39,7 +39,16 @@ class COpenGLCommandPool final : public IGPUCommandPool
             std::unique_lock<std::mutex> lk(mutex);
             mempool.free_n<T>(const_cast<T*>(ptr), n);
         }
-        
+
+        void reset() override
+        {
+            // Free everything in the memory pool from all command buffers
+            // TODO figure out if this is the best way to do it
+            mempool.reset();
```

Comment:
Its probably best, just check it works :)

Reply:
ok, it wont work, you need to be able to iterate every single struct in the `mempool` and run its destructor

Reply:
This is a good point. I remember asking myself "what is wrong here" but couldn't figure out even though I did something [similar to this](https://github.com/Devsh-Graphics-Programming/Nabla/blob/b80d42cc34bc9753fa5b52503843de474fb05e61/include/nbl/video/IGPUCommandBuffer.h#L301) before myself.